langchain/src/llms/bedrock-chat.ts

```diff
@@ -0,0 +1,267 @@
+
+    this.model = fields?.model ?? this.model;
+    const allowedModels = ["ai21", "anthropic", "amazon"];
+    if (!allowedModels.includes(this.model.split(".")[0])) {
+      throw new Error(
+        `Unknown model: '${this.model}', only these are supported: ${allowedModels}`
+      );
+    }
+    const region =
+      fields?.region ?? getEnvironmentVariable("AWS_DEFAULT_REGION");
+    if (!region) {
+      throw new Error(
+        "Please set the AWS_DEFAULT_REGION environment variable or pass it to the constructor as the region field."
+      );
+    }
+    this.region = region;
+    this.credentials = fields?.credentials ?? defaultProvider();
+    this.temperature = fields?.temperature ?? this.temperature;
+    this.maxTokens = fields?.maxTokens ?? this.maxTokens;
+    this.streaming = fields?.streaming ?? this.streaming;
+    this.fetchFn = fields?.fetchFn ?? fetch;
+  }
+
+  /** Call out to Bedrock service model.
+    Arguments:
+      prompt: The prompt to pass into the model.
+
+    Returns:
+      The string generated by the model.
+
+    Example:
+      response = model.call("Tell me a joke.")
+  */
+  async _call(
+    prompt: string,
+    _options: this["ParsedCallOptions"],
+    runManager?: CallbackManagerForLLMRun
+  ): Promise<string> {
+    const provider = this.model.split(".")[0];
+    const service = "bedrock";
+
+    const inputBody = BedrockChatLLMInputOutputAdapter.prepareInput(
+      provider,
+      prompt
+    );
+
+    const url = this.streaming
+      ? new URL(
+          `https://${service}.${this.region}.amazonaws.com/model/${this.model}/invoke-with-response-stream`
+        )
+      : new URL(
+          `https://${service}.${this.region}.amazonaws.com/model/${this.model}/invoke`
+        );
+
+    const request = new HttpRequest({
+      hostname: url.hostname,
+      path: url.pathname,
+      protocol: url.protocol,
+      method: "POST", // method must be uppercase
+      body: JSON.stringify(inputBody),
+      query: Object.fromEntries(url.searchParams.entries()),
+      headers: {
+        // host is required by AWS Signature V4: https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+        host: url.host,
+        accept: "application/json",
+        "Content-Type": "application/json",
+      },
+    });
+
+    const signer = new SignatureV4({
+      credentials: this.credentials,
+      service,
+      region: this.region,
+      sha256: Sha256,
+    });
+
+    const signedRequest = (await signer.sign(request)) as HttpRequest;
+    const response = await this.fetchFn(url, {
+      headers: signedRequest.headers,
+      body: signedRequest.body,
+      method: signedRequest.method,
+    });
+    if (response.status < 200 || response.status >= 300) {
+      throw Error(
+        `Failed to access underlying url '${url}': got ${response.status} ${
+          response.statusText
+        }: ${await response.text()}`
+      );
+    }
+
+    if (this.streaming) {
```

Comment:
We now have a new interface for `.stream()`:

https://js.langchain.com/docs/guides/expression_language/interface

Also, it'd be more elegant to always use the streaming call and implement non-streaming requests as just concatenating the entire stream:

https://github.com/hwchase17/langchainjs/blob/main/langchain/src/llms/ollama.ts#L124